### DISCUSSION TITLE
nothing shows on the screen after instruciton while no error message in console

### DISCUSSION THREAD

**Discussion by xf15** [jspsych/jsPsych#1116] [Q&A] [answered]
can some help me debug and teach me how?

modified https://github.com/jspsych/jsPsych/blob/master/examples/jspsych-audio-keyboard-response.html
```
<!DOCTYPE html>
<html>

<head>
  <script src="../jspsych.js"></script>
  <script src="../plugins/jspsych-audio-keyboard-response.js"></script>
  <script src="../plugins/jspsych-html-button-response.js"></script>
  <link rel="stylesheet" href="../css/jspsych.css">
  </link>
</head>

<body></body>
<script>
  var timeline = [];

  var pre_audio = {
    type: 'html-button-response',
    stimulus: 'Recent versions of Chrome require the user to interact with a page before it can play audio. ' +
      'Clicking the button below counts as an interaction. Be aware of this when planning audio experiments if ' +
      'you want the first trial to include audio.',
    choices: ['Continue']
  }
  timeline.push(pre_audio)

  var stimuli = [{
      file: 'sound/speech_blue.mp3'
    },
    {
      file: 'sound/speech_green.mp3'
    }
  ];
  var trial = {
    type: "audio-keyboard-response",
    stimulus: jsPsych.timelineVariable('file'),
    choices: ['y', 'n'],
    prompt: 'ja'
  }
  var trials = {
    timeline: trial,
    timeline_variables: stimuli,
    randomize_order: true,
    repetitions: 5
  }
  timeline.push(trials)

  jsPsych.init({
    timeline: timeline,
    use_webaudio: false,

  });
</script>

</html>
```

**Comment by becky-gilbert** [comment, marked as answer]
Hi @xf15! One problem is that, in your `trials` procedure, the timeline needs to be an array (see [docs on timeline variables](https://www.jspsych.org/overview/timeline/#timeline-variables)). Sorry that there's no useful error/warning in the console when this happens. We're in the process of adding some more verbose errors for this kind of thing (issue #853).

Another problem has to do with media preloading and the use of timeline variables. Normally media files will be automatically preloaded when you use certain types of plugins, like audio-keyboard-response. But that automatic preloading doesn't happen when the plugin is used in a trial procedure with timeline variables. This issue is in the documentation under [Media Preloading](https://www.jspsych.org/overview/media-preloading/#manual-preloading), but sorry that it's not super easy to find. The solution for this is to preload your audio files manually by using the [`preload_audio` parameter in jsPsych.init](https://www.jspsych.org/core_library/jspsych-core/#parameters_8).

So here's the code that includes these changes:
```javascript
var trials = {
    timeline: [trial],
    timeline_variables: stimuli,
    randomize_order: true,
    repetitions: 5
  }
  timeline.push(trials)

  jsPsych.init({
    timeline: timeline,
    use_webaudio: false,
    preload_audio: ['sound/speech_blue.mp3','sound/speech_green.mp3']
  });
```

**Reply by xf15** [reply]
Thank you @becky-gilbert
